src/chapters/angular-lsn2/event-functions.rst

```diff
@@ -0,0 +1,212 @@
+
+To perform more complicated tasks in response to the user's actions, we can
+call a function when an event occurs. The syntax for this is:
+
+::
+
+   (event) = 'functionName(parameters...)'
+
+Changing the movie list displayed on the webpage requires us to modify the
+``movies`` array in the ``movie-list.component.ts`` file. We will do this by
+creating an ``addMovie`` function and linking it to our event handlers.
+
+Modify the HTML
+----------------
+
+Let's change our code in ``movie-list.component.html`` to call the function
+``addMovie`` and pass the new movie title as the argument.
+
+#. On lines 7 and 8, replace ``true`` with the function call:
+
+   .. sourcecode:: html+ng2
+      :linenos:
+
+      <div class='movies col-4'>
+         <h3>Movies to Watch</h3>
+         <ol>
+            <li *ngFor ="let movie of movies">{{movie}}</li>
+         </ol>
+         <hr>
+         <input #newMovie (keyup.enter)='addMovie(newMovie.value)' type='text' placeholder="Enter Movie Title Here"/>
+         <button (click)='addMovie(newMovie.value)'>Add</button>
+         <p>{{newMovie.value}}</p>
+      </div>
+
+   Now when the user taps "Enter" or clicks the "Add" button after typing, the
+   input ``newMovie.value`` gets sent to the function.
+
+#. Remove the ``<p>{{newMovie.value}}</p>`` from line 9, since the new title
+   should appear on the list instead of below the box.
+
+Define the Function
+--------------------
+
+Open ``movie-list.component.ts`` and examine the code:
+
+.. sourcecode:: TypeScript
+   :linenos:
+
+   import { Component, OnInit } from '@angular/core';
+
+   @Component({
+      selector: 'movie-list',
+      templateUrl: './movie-list.component.html',
+      styleUrls: ['./movie-list.component.css']
+   })
+   export class MovieListComponent implements OnInit {
+      movies = ['Toy Story', 'The Shining', 'Sleepless in Seattle', 'The Martian'];
+
+      constructor() { }
+
+      ngOnInit() {
+      }
+   }
+
+The ``movies`` array stores the titles displayed on the webpage, and we want to
+update this when the user supplies new information.
+
+3. Declare a function called ``addMovie`` that takes one parameter:
+
+   .. sourcecode:: TypeScript
+      :linenos:
+
+      export class MovieListComponent implements OnInit {
+         movies = ['Toy Story', 'The Shining', 'Sleepless in Seattle', 'The Martian'];
+
+         addMovie (newTitle: string) {
+
+         }
+
+         constructor() { }
+
+         ngOnInit() {
+         }
+      }
+
+   Notice that we have to declare the data type for the ``newTitle`` parameter.
+
+#. Now add code to add the new title to the ``movies`` array:
+
+   .. sourcecode:: TypeScript
+      :linenos:
+
+      export class MovieListComponent implements OnInit {
+         movies = ['Toy Story', 'The Shining', 'Sleepless in Seattle', 'The Martian'];
+
+         addMovie (newTitle: string) {
+            this.movies.push(newTitle);
+            return;
```

Comment:
As discussed plz remove all `return` statements that don't return anything.

Reply:
Done.